tests/rptest/tests/pandaproxy_test.py

```diff
@@ -1058,3 +1072,113 @@ def test_fetch_topic(self):
+        assert cc_res['error_code'] == 40101
+
+        cc_res = self._create_consumer(group_id,
+                                       auth=(super_username, super_password))
+        assert cc_res.status_code == requests.codes.ok
+        c0 = Consumer(cc_res.json())
+
+        # Subscribe a consumer
+        self.logger.info(f"Subscribe consumer to topics: {self.topics}")
+        tp_names = [tp.name for tp in self.topics]
+        sc_res = c0.subscribe(tp_names,
+                              auth=(self.username, self.password)).json()
+        assert sc_res['error_code'] == 40101
+
+        sc_res = c0.subscribe(tp_names, auth=(super_username, super_password))
+        assert sc_res.status_code == requests.codes.no_content
+
+        # Fetch from a consumer
+        self.logger.info(f"Consumer fetch")
+        cf_res = c0.fetch(headers=HTTP_CONSUMER_FETCH_JSON_V2_HEADERS,
+                          auth=(self.username, self.password)).json()
+        assert cf_res['error_code'] == 40101
+
+        cf_res = c0.fetch(headers=HTTP_CONSUMER_FETCH_JSON_V2_HEADERS,
+                          auth=(super_username, super_password))
+        assert cf_res.status_code == requests.codes.ok
+        fetch_result = cf_res.json()
+        # 3 topics * 3 msg
+        assert len(fetch_result) == 3 * 3
+        for r in fetch_result:
+            assert r["value"]["object"]
+
+        self.logger.info("Get consumer offsets")
+        offset_data = {
+            'partitions': [{
+                'topic': self.topics[0].name,
+                'partition': 0
+            }]
+        }
+        cof_res = c0.get_offsets(json.dumps(offset_data),
+                                 auth=(self.username, self.password)).json()
+        assert cof_res['error_code'] == 40101
+
+        cof_res = c0.get_offsets(json.dumps(offset_data),
+                                 auth=(super_username, super_password))
+        assert cof_res.status_code == requests.codes.ok
+        offsets = cof_res.json()['offsets']
+        assert len(offsets) == 1
+        assert offsets[0]['topic'] == self.topics[0].name
+        assert offsets[0]['partition'] == 0
+        assert offsets[0]['offset'] == -1
+        assert offsets[0]['metadata'] == ''
+
+        self.logger.info("Set consumer offsets")
+        cos_res = c0.set_offsets(auth=(self.username, self.password)).json()
+        assert cos_res['error_code'] == 40101
+
+        cos_res = c0.set_offsets(auth=(super_username, super_password))
+        assert cos_res.status_code == requests.codes.no_content
+
+        # Remove consumer
+        self.logger.info("Remove consumer")
+        rc_res = c0.remove(auth=(self.username, self.password)).json()
+        assert rc_res['error_code'] == 40101
+
+        rc_res = c0.remove(auth=(super_username, super_password))
+        print(f'Remove Consumer {rc_res.json()}')
+        # assert rc_res.status_code == requests.codes.no_content
```

Comment:
Fixed. Problem was because the request in handlers.cc was std::move'd into a lambda so the default user was passed into the cache instead of the true username.